test/e2e/specs/onboarding/setup__site-assembler.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @group calypso-release
+ */
+
+import {
+	DataHelper,
+	SecretsManager,
+	TestAccount,
+	SiteAssemblerFlow,
+} from '@automattic/calypso-e2e';
+import { Browser, Page } from 'playwright';
+
+declare const browser: Browser;
+
+describe( DataHelper.createSuiteTitle( 'Site Assembler' ), () => {
+	const credentials = SecretsManager.secrets.testAccounts.defaultUser;
+	const siteSlug = credentials.testSites?.primary?.url as string;
+
+	let page: Page;
+
+	beforeAll( async () => {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( 'defaultUser' );
+		await testAccount.authenticate( page );
+	} );
+
+	describe( 'Create a site with the Site Assembler', function () {
+		let startSiteFlow: SiteAssemblerFlow;
+
+		beforeAll( async function () {
+			startSiteFlow = new SiteAssemblerFlow( page );
+		} );
+
+		it( 'Enter Onboarding flow', async function () {
+			await page.goto( DataHelper.getCalypsoURL( '/setup/site-setup', { siteSlug } ), {
+				timeout: 30 * 1000,
+			} );
+		} );
+
+		it( 'Select "Continue" until it lands on the Design Picker', async function () {
+			await startSiteFlow.clickButton( 'Continue' );
+			await startSiteFlow.clickButton( 'Continue' );
+		} );
+
+		it( 'Select "Start designing" and land on the Site Assembler', async function () {
+			await startSiteFlow.clickButton( 'Start designing' );
+			await page.waitForURL(
+				DataHelper.getCalypsoURL( `/setup/site-setup/patternAssembler?siteSlug=${ siteSlug }` ),
+				{
+					timeout: 30 * 1000,
+				}
+			);
+		} );
+
+		it( 'Select "Header"', async function () {
+			await startSiteFlow.selectHeader();
+			await startSiteFlow.clickButton( 'Save' );
+			await page
+				.locator( '.pattern-large-preview__patterns .block-renderer' )
+				.count()
+				.then( ( count ) => {
+					expect( count ).toBe( 1 );
+				} );
+		} );
+
+		it( 'Select "Footer"', async function () {
+			await startSiteFlow.selectFooter();
+			await startSiteFlow.clickButton( 'Save' );
+			await page
+				.locator( '.pattern-large-preview__patterns .block-renderer' )
+				.count()
+				.then( ( count ) => {
+					expect( count ).toBe( 2 );
+				} );
+		} );
+
+		it( 'Click "Continue" and land on the Site Editor', async function () {
+			await startSiteFlow.clickButton( 'Continue' );
+			await page.waitForURL( /.*\/site-editor\/.*/, {
+				timeout: 30 * 1000,
```

Comment:
**suggestion:non-blocking** - I think the`/site-editor` may take longer than 30 seconds to load on the worst days. We can start with this value and see if it leads to flaky runs.